aiida/tools/importexport/common/archive.py

```diff
@@ -252,33 +269,44 @@ def extract_zip(infile, folder, nodes_export_subfolder=None, silent=False):
         with zipfile.ZipFile(infile, 'r', allowZip64=True) as handle:
 
             if not handle.namelist():
-                raise CorruptArchive('no files detected')
+                raise CorruptArchive('no files detected in archive')
+
+            if silent:
+                p_bar = json_files
```

Comment:
what does this mean?
the progress bar is a set of json files?
I hope there is a less dirty way of achieving the behavior you desire.

Reply:
perhaps rather call it "file_iterator" or similar